src/Nova/Slot.php

```diff
@@ -0,0 +1,125 @@
+    }
+
+    public static $group = 'Operations Scheduling';
+
+    public static $defaultAttributes = [
+        'supervision_id' => 2,
+    ];
+
+    public function fieldsForIndex(NovaRequest $request)
+    {
+        return array_filter([
+            ID::make()->sortable(),
+            Text::make('Room', 'room.id', function () {
+                return $this->room->name;
+            })->sortable(),
+            Text::make('Start', 'start_at', function () {
+                return $this->formatted_start;
+            })->sortable(),
+            Date::make('Date', 'date')->sortable(),
+            Number::make('Participants'),
+            Number::make('Available', 'participants_available'),
+        ]);
+    }
+
+    public function fields(Request $request)
+    {
+        return array_filter([
+            Text::make('Title')->exceptOnForms(),
+            nova('room') ? BelongsTo::make('Room', 'room', nova('room'))->hideWhenUpdating()->required() : null,
+            Text::make('Game Start', 'start_at', function () {
+                return $this->formatted_start;
+            })->exceptOnForms(),
+            DateTime::make('Start At')->required(),
+            nova('rate') ? BelongsTo::make('Rate', 'rate', nova('rate'))->hideWhenCreating()->nullable() : null,
+            nova('supervision') ? BelongsTo::make('Supervision', 'supervision', nova('supervision'))->hideWhenCreating()->nullable() : null,
+            new Panel('Participant Details', $this->participantFields()),
+            nova('booking') ? HasMany::make('Bookings', 'bookings', nova('booking')) : null,
+            nova('block') ? HasMany::make('Blocks', 'blocks', nova('block')) : null,
+            nova('note') ? MorphMany::make('Notes', 'notes', nova('note')) : null,
+            nova('game') ? HasOne::make('Game', 'game', nova('game'))->nullable()->exceptOnForms() : null,
+
+            new Panel('Data Fields', $this->dataFields()),
+        ]);
+    }
+
+    protected function participantFields()
+    {
+        return [
+            Number::make('Participants')->exceptOnForms(),
+            Number::make('Participants Blocked')->exceptOnForms(),
+            Number::make('Participants Available')->exceptOnForms(),
+            Date::make('Date')->exceptOnForms(),
+            DateTime::make('End At')->exceptOnForms(),
+            DateTime::make('Room Available At')->exceptOnForms(),
+        ];
+    }
+
+    protected function dataFields(): array
+    {
+        return array_filter([
+            ID::make(),
+            Text::make('Slot Number')->exceptOnForms(),
+            DateTime::make('Created At')->exceptOnForms(),
+            nova('user') ? BelongsTo::make('Updated By', 'updater', nova('user'))->exceptOnForms() : null,
+            DateTime::make('Updated At')->exceptOnForms(),
+        ]);
+    }
+
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\FutureSlots,
```

Comment:
@drewroberts can you let me know what packages contain these filters so I can refactor them?

Reply:
This scheduling package